app_build_suite/build_steps/helm.py

```diff
@@ -276,6 +278,82 @@ def run(self, config: argparse.Namespace, _: Context) -> None:
             raise BuildError(self.name, "Linting failed")
 
 
+class HelmRequirementsUpdater(BuildStep):
+    """
+    Executes helm dependency update.
+    """
+
+    _helm_bin = "helm"
+    _min_helm_version = "3.2.0"
+    _max_helm_version = "4.0.0"
+
+    @property
+    def steps_provided(self) -> Set[StepType]:
+        return {STEP_BUILD}
+
+    # noinspection PyMethodMayBeStatic
+    def _should_run(self, config: argparse.Namespace) -> bool:
+        return config.replace_chart_version_with_git
+
+    # noinspection PyMethodMayBeStatic
+    def _chart_lock_exists(self, config: argparse.Namespace) -> bool:
+        return os.path.isfile(os.path.join(config.chart_dir, _chart_lock))
+
+    def pre_run(self, config: argparse.Namespace) -> None:
+        """
+        Checks if the required version of helm is installed and if the lock file is present.
+        :param config: the config object
+        :return: None
+        """
+        if not self._should_run(config):
+            logger.debug("No chart version override requested, skipping dependency update.")
+            return
+        if not self._chart_lock_exists(config):
```

Comment:
The same validation is done for the helm build step. I think we should extract this to some helper method/class (like abstract `HelmBasedBuildStep` class) that includes this method to avoid duplicating code.

Reply:
I think this is not as easy as you think. I think that would introduce more abstraction than required.

Correct me if I'm wrong. You mean the `_is_enabled` function of `HelmGitVersionSetter` compares to the `_should_run` function here in `HelmRequirementsUpdater`? Please note the first checks for both `config.replace_chart_version_with_git or config.replace_app_version_with_git` while the latter only checks `config.replace_chart_version_with_git`. Also the `_chart_lock_exists` function is freshly introduced in this PR. No other step checked for existende of a `Chart.lock` file.